scripts/ci/pairing/pkg/cmd/pair.go

```diff
@@ -0,0 +1,176 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+
+	"github.com/codeready-toolchain/pairing/pkg/cmd/flags"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/google/go-github/v44/github"
+	"github.com/spf13/cobra"
+)
+
+type PairingServiceInterface interface {
+	shouldPair(orgForPairing, repoForPairing, currentRemoteName, currentBranchName string) (bool, error)
+}
+
+type PairingService struct{}
+
+func (s *PairingService) shouldPair(orgForPairing, repoForPairing, currentRemoteName, currentBranchName string) (bool, error) {
+	return shouldPair(orgForPairing, repoForPairing, currentRemoteName, currentBranchName)
+}
+
+func NewPairCmd() *cobra.Command {
+	var cloneDir, organization, repository string
+
+	command := &cobra.Command{
+		Use:   "pair --clone-dir=<clone-dir> --organization=<organization> --repository=<repository>",
+		Short: "Pair PRs in CI between the given organization and repository into the given clone directory",
+		Long:  `Automatically tries to pair a PR opened on a specific repository with a branch of the same name that potentially could exist in the given organization and repository.`,
+		Args:  cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return Pair(cloneDir, organization, repository, &PairingService{})
+		},
+	}
+
+	command.Flags().StringVarP(&cloneDir, "clone-dir", "c", "", "Directory to clone into")
+	flags.MustMarkRequired(command, "clone-dir")
+
+	command.Flags().StringVarP(&organization, "organization", "o", "", "Organization to pair")
+	flags.MustMarkRequired(command, "organization")
+
+	command.Flags().StringVarP(&repository, "repository", "r", "", "Repository to pair")
+	flags.MustMarkRequired(command, "repository")
+
+	return command
+}
+
+// listOpenPRs lists open pull requests for the given repository
+func listOpenPRs(owner, repo string) ([]*github.PullRequest, error) {
+	client := github.NewClient(nil) // no authentication needed
```

Comment:
I would be very careful with using unauthenticated API - we could easily hit the rate limit. This is also why the existing pairing mechanism doesn't rely on the GH API at all and gets the existing branches in the forked repo via sending request to `/info/refs?service=git-upload-pack`
see: https://github.com/codeready-toolchain/host-operator/blob/a89c3f20a7ae31d0093587d8b54848b4c7ebbc50/make/test.mk#L110